src/Plugin/Field/FieldWidget/LocalistUrlWidget.php

```diff
@@ -0,0 +1,444 @@
+    if ($item->uri) {
+      parse_str(parse_url(urldecode($item->uri), PHP_URL_QUERY), $query_parameters);
+    }
+
+    $element['filters']['type'] = $this->getFilters($query_parameters['type'] ?? '');
+    $element['filters']['type']['event_audience']['#chosen'] = TRUE;
+    $element['filters']['type']['event_subject']['#chosen'] = TRUE;
+    $element['filters']['type']['event_types']['#chosen'] = TRUE;
+    $element['filters']['group_id'] = $this->getGroups($query_parameters['group_id'] ?? '');
+    $element['filters']['venue_id'] = $this->getPlaces($query_parameters['venue_id'] ?? '');
+
+    // It is not yet possible to set "Featured" or "Sponsored"
+    // events in the Localist UI.
+    // I'm going to comment out these fields for now, but
+    // if the Localist UI is updated in the future, we can
+    // re-enable them easily here.
+    /*
+    $element['filters']['picks'] = [
+    '#type' => 'checkbox',
+    '#title' => $this->t('Only Show Featured'),
+    '#default_value' => $query_parameters['picks'] ?? FALSE,
+    ];
+    $element['filters']['sponsored'] = [
+    '#type' => 'checkbox',
+    '#title' => $this->t('Only Show Sponsored'),
+    '#default_value' => $query_parameters['sponsored'] ?? FALSE,
+    ];
+     */
+    // End of unused form elements.
+    $element['filters']['match'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Content Must Match'),
+      '#default_value' => $query_parameters['match'] ?? NULL,
+      '#empty_option' => $this->t('At least one place or group, and one filter item'),
+      '#options' => [
+        'any' => $this->t('Any venue, group, or filter item'),
+        'all' => $this->t('At least one venue and group, and all filter items'),
+        'or' => $this->t('Any venue or group, and one filter item'),
+      ],
+
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function massageFormValues(array $values, array $form, FormStateInterface $form_state) {
+
+    foreach ($values as $delta => &$value) {
+
+      $types_querystring = '';
+
+      foreach ($value['filters'] as $key => &$filter_values) {
+        if (is_array($filter_values)) {
+          $filter_values = self::flattenValues($filter_values);
+          foreach ($filter_values as $type) {
+            $types_querystring .= '&type[]=' . $type;
+          }
+          unset($value['filters'][$key]);
+        }
+      }
+
+      $value['filters'] = array_filter($value['filters']);
+
+      if (empty($value['filters'])) {
+        unset($values[$delta]);
+        continue;
+      }
+
+      $value['filters']['days'] = '365';
+
+      // We may in the future have a configuration value
+      // to include the "distinct"key to our API call.
+      // This tries to find such a value,
+      // and applies the key if it finds it.
+      try {
+        $config = \Drupal::config('config_pages.type.stanford_events_importer');
+        if ($config->get('third_party_settings.localist_select_distinct')) {
```

Comment:
This should be on the field widget settings, not third party settings. The widget will then build the URL based on that setting, so the distinct gets saved in the database directly with the rest of the url. We don't want the user to be able to change the option. Put int into the `settingsForm` & `defaultSettings` methods. Also this module should never rely on any config pages.

Reply:
fixed.  I put a checkbox in the field settings form as well.